topdown/reachable.go

```diff
@@ -72,43 +72,59 @@ func builtinReachable(bctx BuiltinContext, operands []*ast.Term, iter func(*ast.
 	return iter(ast.NewTerm(reached))
 }
 
-// pathBuilder is called recursively to build an array of paths that are reachable from the root
-func pathBuilder(graph ast.Object, root *ast.Term, path []*ast.Term, paths []*ast.Term, reached ast.Set) []*ast.Term {
+// pathBuilder is called recursively to build a Set of paths that are reachable from the root
+func pathBuilder(graph ast.Object, root *ast.Term, path []*ast.Term, edgeRslt ast.Set, reached ast.Set) ast.Set {
+	paths:= []*ast.Term{}
+
 	if edges := graph.Get(root); edges != nil {
 		path = append(path, root)
 
 		if numberOfEdges(edges) >= 1 {
+
 			foreachVertex(edges, func(neighbor *ast.Term) {
+
 				if reached.Contains(neighbor) {
 					// If we've already reached this node, return current path (avoid infinite recursion)
 					paths = append(paths, path...)
+					edgeRslt.Add(ast.ArrayTerm(paths...))
 				} else {
 					reached.Add(root)
-					paths = pathBuilder(graph, neighbor, path, paths, reached)
+					pathBuilder(graph, neighbor, path, edgeRslt, reached)
+
 				}
+
+
 			})
+
 		} else {
 			paths = append(paths, path...)
+			edgeRslt.Add(ast.ArrayTerm(paths...))
+
+
 		}
 	} else {
 		// Node is nonexistent (not in graph). Commit the current path (without adding this root)
 		paths = append(paths, path...)
+		edgeRslt.Add(ast.ArrayTerm(paths...))
+
+
 	}
+	return edgeRslt
```

Comment:
🤔 What's the point in returning anything when the function adds to this _argument_ set anyways...? I might be missing something but it looks like this could be
```go
func pathBuilder(graph ast.Object, root *ast.Term, path []*ast.Term, edgeRslt ast.Set, reached ast.Set)
```
where `edgeRslt` is updated in function body.